src/workers/block_listenser.rs

```diff
@@ -296,94 +294,60 @@ impl BlockListener {
-            let postgres = postgres.clone();
-            let slots_task_queue = slots_task_queue.clone();
-
-            tokio::spawn(async move {
-                let slots_task_queue = slots_task_queue.clone();
-                loop {
-                    let (slot, error_count) = {
-                        let mut queue = slots_task_queue.lock().await;
-                        match queue.pop_front() {
-                            Some(t) => t,
-                            None => {
-                                tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
-                                continue;
-                            }
-                        }
-                    };
+        tokio::spawn(async move {
+            let mut latest_slot = self
+                .rpc_client
+                .get_slot_with_commitment(commitment_config)
+                .await?;
 
-                    if error_count > 10 {
-                        // retried for 10 times / there should be no block for this slot
-                        warn!(
-                            "unable to get block at slot {} and commitment {}",
-                            slot, commitment_config.commitment
-                        );
-                        continue;
-                    }
+            info!("Listening to blocks {commitment_config:?}");
 
-                    if let Err(_) = this
-                        .index_slot(slot, commitment_config, postgres.clone())
-                        .await
-                    {
-                        // usually as we index all the slots even if they are not been processed we get some errors for slot
-                        // as they are not in long term storage of the rpc // we check 10 times before ignoring the slot
-                        tokio::time::sleep(tokio::time::Duration::from_millis(50)).await;
-                        {
-                            let mut queue = slots_task_queue.lock().await;
-                            queue.push_back((slot, error_count + 1));
-                        }
-                    };
-                    //   println!("{i} thread done slot {slot}");
-                }
-            });
-        }
+            let mut slot_que = Vec::new();
 
-        let rpc_client = self.rpc_client.clone();
-        tokio::spawn(async move {
-            let slots_task_queue = slots_task_queue.clone();
-            let last_latest_slot = self
-                .block_store
-                .get_latest_block_info(commitment_config)
-                .await
-                .slot;
-            // -5 for warmup
-            let mut last_latest_slot = last_latest_slot - 5;
-
-            // storage for recent slots processed
-            let rpc_client = rpc_client.clone();
             loop {
-                let new_slot = rpc_client
-                    .get_slot_with_commitment(commitment_config)
+                let mut new_block_slots = self
```

Comment:
You can remove get blocks and move to the previous implementation of get slot and then create a range of last_slot to current slot. 
Some time i feel getBlocks does not work very well.

Reply:
Fixed c5ef5a0cf72ccd7865944dbcd9fbda58729dd9a3